### PR TITLE
Merge EditorContainer into EditCell + cleanups

### PR DESCRIPTION
**Pull request by nstepien**
-1 div!

### CODE REVIEW

File: src/EditCell.tsx [hunk 1]
```diff
@@ -1,37 +1,38 @@
...
-import type { CellRendererProps, SharedEditorProps, Omit } from './types';
+import type { CellRendererProps, SharedEditorProps } from './types';
 
 const cellEditing = css`
   padding: 0;
 `;
 
-const cellEditingClassname = `rdg-cell-editing ${cellEditing}`;
+// TODO: remove 1 of the two classes
+const cellEditingClassname = `rdg-cell-editing rdg-editor-container ${cellEditing}`;
```
Comment (nstepien) [inline src/EditCell.tsx:14]: WDYT? We have some internal code depending on both class names at the moment.

File: src/EditCell.tsx [hunk 2]
```diff
@@ -1,37 +1,38 @@
...
   CellRendererProps<R, SR>,
   'rowIdx' | 'row' | 'column' | 'colSpan'
 >;
 
-interface EditCellProps<R, SR>
-  extends SharedCellRendererProps<R, SR>,
-    Omit<React.HTMLAttributes<HTMLDivElement>, 'style' | 'children'> {
+interface EditCellProps<R, SR> extends SharedCellRendererProps<R, SR> {
   editorProps: SharedEditorProps<R>;
+  onKeyDown: Required<React.HTMLAttributes<HTMLDivElement>>['onKeyDown'];
```
Comment (nstepien) [inline src/EditCell.tsx:23]: `EditCell` is a strictly internal component with only specific props.

File: src/EditCell.tsx [hunk 3]
```diff
@@ -41,30 +42,31 @@ export default function EditCell<R, SR>({
...
   );
 
-  function getCellContent() {
-    if (dimensions === null) return;
-    const { scrollTop: docTop, scrollLeft: docLeft } =
-      document.scrollingElement ?? document.documentElement;
-    const { left, top } = dimensions;
-    const gridLeft = left + docLeft;
-    const gridTop = top + docTop;
+  let content;
```
Comment (nstepien) [inline src/EditCell.tsx:51]: I prefer doing this over having a function that is only ever called once, unconditionally.

File: src/EditCell.tsx [hunk 4]
```diff
@@ -1,37 +1,38 @@
...
   column,
   colSpan,
   row,
   rowIdx,
-  editorProps,
-  ...props
+  onKeyDown,
+  editorProps
 }: EditCellProps<R, SR>) {
   const [dimensions, setDimensions] = useState<{ left: number; top: number } | null>(null);
```
Comment (nstepien) [inline src/EditCell.tsx:34]: Really wonder if we still need this

Comment (amanmahajan7) [reply src/EditCell.tsx:34]: I think this was needed when we were rendering the whole editor in a portal. I am okay with removing them add them back if needed.

File: src/EditCell.tsx (hunk 2 above)
Comment (amanmahajan7) [reply src/EditCell.tsx:23]: Should we also flatten the props and removed `editorProps`?

File: src/EditCell.tsx (hunk 1 above)
Comment (amanmahajan7) [reply src/EditCell.tsx:14]: I think we can remove one if possible

**Review by amanmahajan7 — DISMISSED** [review]

File: stories/demos/components/Editors/SelectEditor.tsx [hunk 5]
```diff
@@ -27,13 +20,13 @@ export function SelectEditor({
       styles={{
         control: (provided) => ({
           ...provided,
-          height: rowHeight - 1,
+          height: 'var(--row-height)',
```
Comment (amanmahajan7) [inline stories/demos/components/Editors/SelectEditor.tsx:23]:
```suggestion
          height: 'calc(var(--row-height) - 1)',
```

image

We can probably remove this editor completely

**Review by amanmahajan7 — DISMISSED** [review]

**Review by amanmahajan7 — APPROVED** [review]

**Pull request merged**